src/Block/Fetch.hs

```diff
@@ -66,67 +72,99 @@ data OgmiosInfo = OgmiosInfo
 
 newtype OgmiosWorkerMVar = MkOgmiosWorkerMVar (MVar (Async ()))
 
+newtype ControlApiToken = ControlApiToken (Maybe String)
+  deriving stock (Eq, Show)
+
+withControlApiTokenToken ::
```

Comment:
Verification of auth token is part of an API, not block fetcher and it should be validated there. There shouldn't be any required changes to block fetcher logic after adding authorization in REST/WS API